### ISSUE TITLE
Include jni plugin while installing Elektra

### ISSUE THREAD

**Issue by dmoisej** [ElektraInitiative/libelektra#2790]
## Steps to Reproduce the Problem

The guide: https://www.libelektra.org/plugins/jni
- `git clone https://github.com/ElektraInitiative/libelektra.git`
- `cd libelektra`
- `mkdir build && cd build`
- `cmake -DPLUGINS="ALL;-EXPERIMENTAL;jni" ..`

## Expected Result

The output of  `cmake` command:
-- Looking for JNI_VERSION_1_8 - found
-- Include Plugin jni for shared builds

## Actual Result

The output of  `cmake` command:
-- Looking for JNI_VERSION_1_8 - not found
-- Exclude Plugin jni because jni.h does not define JNI_VERSION_1_8

## System Information

- Elektra Version: master
- Operating System or Docker Container?: Ubuntu 16.06
- Versions of other relevant software?

## Further Log Files and Output
[CMakeError.log](https://github.com/ElektraInitiative/libelektra/files/3294261/CMakeError.log)
[CMakeOutput.log](https://github.com/ElektraInitiative/libelektra/files/3294262/CMakeOutput.log)

## Additional Information

- `printenv`

> JAVA_HOME=/usr/lib/jvm/java-8-oracle/

![image](https://user-images.githubusercontent.com/32390104/59566196-e717d300-905c-11e9-8cf3-e9176a5e93df.png)

**Comment by dmoisej** [comment]
@markus2330 
I have recreated build folder and run cmake command. The output files are attached to the issue description.

**Comment by markus2330** [comment]
Thank you for creating this issue!

The CMakeError.log you linked above says:
```
/home/dmoisej/Projects/dmoisej/libelektra/build/CMakeFiles/CMakeTmp/CheckSymbolExists.c:2:10: fatal error: 'jni.h' file not found
#include <jni.h>
         ^
```

It seems like the header file cannot be found on your system.

So the header file seems to be not in DIRS: ${JNI_INCLUDE_DIRS} ${JAVA_INCLUDE_PATH} ${JAVA_INCLUDE_PATH2} ${JAVA_AWT_INCLUDE_PATH}

Can you please tell us what DIRS contains and where jni.h actually is?

**Comment by dmoisej** [comment]
@markus2330 
with the `message()` method, I've got the following result
![image](https://user-images.githubusercontent.com/32390104/59566489-47a90f00-9061-11e9-8a8d-598d3dc58be7.png)
![image](https://user-images.githubusercontent.com/32390104/59566491-51cb0d80-9061-11e9-9443-c4368a3222e3.png)

**Comment by dmoisej** [comment]
@markus2330 I see, that my JNI_INCLUDE_PATH is not correct. Is this system variable or where can I respecify it?

**Comment by markus2330** [comment]
JNI_INCLUDE_DIRS (and JAVA_INCLUDE_PATH     = the include path to jni.h) should be set by the CMake package JNI (`find_package (JNI QUIET)`). The package should be located in /usr/share/cmake-*/Modules/FindJNI.cmake

As the docu of the module says: "The caller may set variable JAVA_HOME to specify a Java installation prefix explicitly". Afaik, there is no other easy way to influence this script except of directly modifying CMake source (and simply setting the correct path).

You can try to set cmake variables with -D but the script will overwrite it later.

It would be nice if you can describe this in a tutorial.

**Comment by dmoisej** [comment]
@markus2330 so according to email, you want to say, that the problem is because I am using now new enough version of Ubuntu?

**Comment by markus2330** [comment]
I wanted to say that this problem most likely is specific to the operating system (or more concrete: to a mismatch of FindJNI.cmake and the installed Java) and I am not sure if it is a good idea to put a lot of time into it.

But to summarize the main things in a tutorial is definitely a good idea (how to debug it, how to set JAVA_HOME and one example of how it finally works)

**Comment by dmoisej** [comment]
@markus2330 understood, thanks for info.

**Comment by sanssecours** [comment]
I looked into the issue. If somebody else wants to debug the problem, just use the [Docker image for Debian sid](https://github.com/ElektraInitiative/libelektra/blob/master/scripts/docker/debian/sid/Dockerfile) or look at the current `cmake` output of [`debian-unstable-full`](https://build.libelektra.org/jenkins/blue/organizations/jenkins/libelektra/detail/master/761/pipeline/42). As far as I can tell, the `RESET` part of

https://github.com/ElektraInitiative/libelektra/blob/52a153007b6a083ab2017b76f13ca32290882e49/cmake/Modules/SafeCheckSymbolExists.cmake#L23

seems to be the problem. Maybe it would make sense, if you just remove `RESET` in PR #2784 Dmytro.

**Comment by dmoisej** [comment]
@sanssecours thank you for information. Will do a little bit later today. 

Just to clarify, do I have to remove the whole line `cmake_push_check_state (RESET)` or should I remove RESET as the argument?

**Comment by sanssecours** [comment]
> @sanssecours thank you for information. Will do a little bit later today.
> 
> Just to clarify, do I have to remove the whole line `cmake_push_check_state (RESET)` or should I remove RESET as the argument?

Just removing the `RESET` argument should be enough.

**Comment by markus2330** [comment]
fixed in  #2784
